### ISSUE TITLE
Restart to update does not work correct from SSH workspace

### ISSUE THREAD

**Issue by fbehrens** [microsoft/vscode-remote-release#3582]
when git the message `Restart to update` from my SSH connected workspace,

the update works fine, but it does not restart the same workspace (just an emtpy new one)

I have experienced this many times on the recent updates,

I am not bothered much, but report it, in case someone likes to fix it.

![image](https://user-images.githubusercontent.com/13704/91278187-2d4de380-e784-11ea-9190-a4292a92aaf6.png)

**Comment by roblourens** [comment]
I'm not sure why this would happen @fbehrens. Does it reopen the workspace when you run the "Reload Window" command (with no update)?

**Comment by fbehrens** [comment]
Yes!

 `Developer: Reload Window` works fine, and does reconnect to the SSH workspace

**Comment by roblourens** [comment]
When this happens, are you saying that it opens an empty window on the same SSH remote, or it opens an empty window that is not connected to the remote?

When this happens next, can you check the dev tools for errors (Help > Toggle Developer Tools)? Also have you set this setting: `window.restoreWindows`?

**Comment by fbehrens** [comment]
After update there is an empty window not connected to SSH remote (local workspace)

The setting is `"window.restoreWindows": "all"`.

Will open developer tools on next update, (also record the update via ScreenToGif).

**Comment by fbehrens** [comment]
Have just recorded the gif, but not looked at developer tools 🤦 . 

![update vscode with ssh](https://user-images.githubusercontent.com/13704/92885948-8455e880-f413-11ea-96c0-4e81b5fff760.gif)

Will do next time 🙏

**Comment by fbehrens** [comment]
Did this time and see no suspicious entries on the developer toos console 
```
abstractExtensionService.ts:505 Extension 'ms-vscode.powershell-preview cannot use PROPOSED API (must started out of dev or enabled via --enable-proposed-api)
updateEnableProposedApi @ abstractExtensionService.ts:493
abstractExtensionService.ts:420 [ms-vscode.powershell-preview]: Command `workbench.action.debug.start` appears multiple times in the `commands` section.
abstractExtensionService.ts:420 [ms-vscode.powershell-preview]: Command `workbench.action.closePanel` appears multiple times in the `commands` section.
abstractExtensionService.ts:420 [ms-vscode.powershell-preview]: Command `workbench.action.positionPanelLeft` appears multiple times in the `commands` section.
abstractExtensionService.ts:420 [ms-vscode.powershell-preview]: Command `workbench.action.positionPanelBottom` appears multiple times in the `commands` section.
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope source.sql.
Old grammar file: file:///c%3A/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/extensions/sql/syntaxes/sql.tmLanguage.json.
New grammar file: file:///c%3A/Users/imbeh/.vscode/extensions/ms-mssql.mssql-1.9.0/syntaxes/SQL.plist
register @ TMScopeRegistry.ts:43
console.ts:137 [Extension Host] (node:17812) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ console.ts:70
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/semver-umd.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/index.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/index.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/ExperimentationService.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/TasApiFeatureProvider.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/FilteredFeatureProvider.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/BaseFeatureProvider.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/AxiosHttpClient.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/ExperimentationServiceAutoPolling.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/ExperimentationServiceBase.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/MemoryKeyValueStorage.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/PollingService.js.map: 
DevTools failed to load SourceMap: Could not load content for file:///C:/Users/imbeh/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/main.js.map: 

```

**Comment by roblourens** [comment]
I tried a couple more times, with that setting set, and can't repro. Any ideas @bpasero?

**Comment by bpasero** [comment]
@roblourens I think @joaomoreno owns the update mechanism on Windows but I think all we do is simply start the Code executable again after the update finishes, so in that regard it should not be any different from just restarting normally.

Maybe this is an issue with the remote server needing an update as well and that maybe has a race condition and we fallback to an empty window? Maybe @alexdima has more ideas.

**Comment by alexdima** [comment]
If this would be a problem with the remote side there would be some kind of error. I believe the problem is that after an update, vs code restarts with an empty window and not with the same workspace.

**Comment by bpasero** [comment]
Unless there is a setting that drives this, I cannot really explain it. We persist state on shutdown that we need to restart but I assume we do a clean exit before applying the update.

**Comment by bpasero** [comment]
@aeschli could https://github.com/microsoft/vscode/issues/107111 be the cause of this issue too? We fail to restore windows after restart?

**Comment by fbehrens** [comment]
I have a new update pending.

Can I do something for you which helps to identify the issue ?

**Comment by roblourens** [comment]
I will optimistically call this a dupe of https://github.com/microsoft/vscode/issues/107111. We will see whether it is fixed in 1.50
